cluster-controller/src/test/java/io/strimzi/controller/cluster/resources/KafkaConnectS2IClusterTest.java

```diff
@@ -0,0 +1,388 @@
+
+    @Test
+    public void testGenerateService()   {
+        Service svc = kc.generateService();
+
+        assertEquals("ClusterIP", svc.getSpec().getType());
+        assertEquals(ResourceUtils.labels("strimzi.io/cluster", cluster, "strimzi.io/type", "kafka-connect-s2i", "strimzi.io/kind", "cluster", "strimzi.io/name", kc.kafkaConnectClusterName(cluster)), svc.getMetadata().getLabels());
+        assertEquals(ResourceUtils.labels("strimzi.io/cluster", cluster, "strimzi.io/type", "kafka-connect-s2i", "strimzi.io/kind", "cluster", "strimzi.io/name", kc.kafkaConnectClusterName(cluster)), svc.getSpec().getSelector());
+        assertEquals(1, svc.getSpec().getPorts().size());
+        assertEquals(new Integer(KafkaConnectCluster.REST_API_PORT), svc.getSpec().getPorts().get(0).getPort());
+        assertEquals(KafkaConnectCluster.REST_API_PORT_NAME, svc.getSpec().getPorts().get(0).getName());
+        assertEquals("TCP", svc.getSpec().getPorts().get(0).getProtocol());
+    }
+
+    @Test
+    public void testPatchService()   {
+        Service orig = new ServiceBuilder()
+                .withNewMetadata()
+                    .withName(kc.kafkaConnectClusterName(cluster))
+                    .withNamespace(namespace)
+                .endMetadata()
+                .withNewSpec()
+                    .withType("ClusterIP")
+                .endSpec()
+                .build();
+
+        Service svc = kc.patchService(orig);
+
+        assertEquals(ResourceUtils.labels("strimzi.io/cluster", cluster, "strimzi.io/type", "kafka-connect-s2i", "strimzi.io/kind", "cluster", "strimzi.io/name", kc.kafkaConnectClusterName(cluster)), svc.getMetadata().getLabels());
+        assertEquals(ResourceUtils.labels("strimzi.io/cluster", cluster, "strimzi.io/type", "kafka-connect-s2i", "strimzi.io/kind", "cluster", "strimzi.io/name", kc.kafkaConnectClusterName(cluster)), svc.getSpec().getSelector());
+    }
+
+    @Test
+    public void testGenerateDeploymentConfig()   {
+        DeploymentConfig dep = kc.generateDeploymentConfig();
+
+        assertEquals(kc.kafkaConnectClusterName(cluster), dep.getMetadata().getName());
+        assertEquals(namespace, dep.getMetadata().getNamespace());
+        assertEquals(ResourceUtils.labels("strimzi.io/cluster", cluster, "strimzi.io/type", "kafka-connect-s2i", "strimzi.io/kind", "cluster", "strimzi.io/name", kc.kafkaConnectClusterName(cluster)), dep.getMetadata().getLabels());
+        assertEquals(new Integer(replicas), dep.getSpec().getReplicas());
+        assertEquals(ResourceUtils.labels("strimzi.io/cluster", cluster, "strimzi.io/type", "kafka-connect-s2i", "strimzi.io/kind", "cluster", "strimzi.io/name", kc.kafkaConnectClusterName(cluster)), dep.getSpec().getTemplate().getMetadata().getLabels());
+        assertEquals(1, dep.getSpec().getTemplate().getSpec().getContainers().size());
+        assertEquals(kc.kafkaConnectClusterName(cluster), dep.getSpec().getTemplate().getSpec().getContainers().get(0).getName());
```

Comment:
I would use a local for `dep.getSpec().getTemplate().getSpec().getContainers().get(0)` because you're repeating it a lot.